sklearn/model_selection/tests/test_validation.py

```diff
@@ -1299,3 +1302,19 @@ def test_permutation_test_score_pandas():
         check_series = lambda x: isinstance(x, TargetType)
         clf = CheckingClassifier(check_X=check_df, check_y=check_series)
         permutation_test_score(clf, X_df, y_ser)
+
+
+def test_cross_val_score_labels_paramter():
```

Comment:
I could create that test. But I'm not sure how I'm checking the change I made here. There was already an option to pass a subset of classes in f1_score so I believe thats working.
Regarding passing a class which doesn't exist in the train/test set, I'm not sure if f1_score is even defined for that class. Right now, sklearn API flags a warning in such cases and makes the score 0 for that class. I couldn't find supporting literature for that though, is it a theoretical concept or more like a business decision?